test/csuite/scope/main.c

```diff
@@ -0,0 +1,243 @@
+		 * or failure to copy application information into library
+		 * memory.
+		 */
+		if (recno)
+			cursor->set_key(cursor, (uint64_t)1);
+		else {
+			strcpy(keybuf, KEY);
+			cursor->set_key(cursor, keybuf);
+		}
+		strcpy(valuebuf, VALUE);
+		cursor->set_value(cursor, valuebuf);
+
+		/*
+		 * The application must keep key and value memory valid until
+		 * the next operation that positions the cursor, modifies the
+		 * data, or resets or closes the cursor.
+		 *
+		 * Modifying either the key or value buffers is not permitted.
+		 */
+		switch (op->func) {
+		case INSERT:
+			testutil_check(cursor->insert(cursor));
+			break;
+		case SEARCH:
+			testutil_check(cursor->search(cursor));
+			break;
+		case SEARCH_NEAR:
+			testutil_check(cursor->search_near(cursor, &exact));
+			break;
+		case REMOVE:
+			testutil_check(cursor->remove(cursor));
+			break;
+		case RESERVE:
+#if 0
+			testutil_check(cursor->reserve(cursor));
+#endif
+			break;
+		case UPDATE:
+			testutil_check(cursor->update(cursor));
+			break;
+		}
+
+		/*
+		 * The cursor should no longer reference application memory,
+		 * and application buffers can be safely overwritten.
+		 */
+		memset(keybuf, 'K', sizeof(keybuf));
+		memset(valuebuf, 'V', sizeof(valuebuf));
+
+		/*
+		 * Check that get_key/get_value behave as expected after the
+		 * operation.
+		 */
+		switch (op->func) {
+		case INSERT:
+		case REMOVE:
+			/*
+			 * Insert and remove configured with a search key do
+			 * not position the cursor and have no key or value.
+			 *
+			 * There should be two error messages, ignore them.
+			 */
+			ignore_errors = 2;
+			if (recno)
+				testutil_assert(
+				    cursor->get_key(cursor, &keyr) != 0);
+			else
+				testutil_assert(
+				    cursor->get_key(cursor, &key) != 0);
+			testutil_assert(cursor->get_value(cursor, &value) != 0);
+			testutil_assert(ignore_errors == 0);
+			break;
+		case RESERVE:
+		case SEARCH:
+		case SEARCH_NEAR:
+		case UPDATE:
+			/*
+			 * Reserve, search, search-near and update position the
+			 * cursor and have both a key and value.
+			 *
+			 * Any key/value should not reference application
+			 * memory.
+			 */
+			if (recno) {
+				testutil_assert(
+				    cursor->get_key(cursor, &keyr) == 0);
+				testutil_assert(keyr == 1);
+			} else {
+				testutil_assert(
+				    cursor->get_key(cursor, &key) == 0);
+				testutil_assert(key != keybuf);
+				testutil_assert(strcmp(key, KEY) == 0);
+			}
+			testutil_assert(cursor->get_value(cursor, &value) == 0);
+			testutil_assert(value != valuebuf);
+			testutil_assert(strcmp(value, VALUE) == 0);
+			break;
+		}
+	}
+}
+
+static void
+run(WT_CONNECTION *conn, const char *uri, const char *config)
+{
+	WT_SESSION *session;
+
+	testutil_check(conn->open_session(conn, NULL, NULL, &session));
+	testutil_check(session->create(session, uri, config));
+	cursor_scope_ops(session, uri);
+	testutil_check(session->close(session, NULL));
+}
+
+int
+main(int argc, char *argv[])
```

Comment:
I don't feel strongly: I know you've converted a C example into this test, but is there any reason this can't be Python?